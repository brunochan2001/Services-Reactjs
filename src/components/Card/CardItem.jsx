import React from 'react';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import Divider from '@material-ui/core/Divider';
import Typography from '@material-ui/core/Typography';
import DeleteIcon from '@mui/icons-material/Delete';
import EditIcon from '@mui/icons-material/Edit';
import Stack from '@mui/material/Stack';
import Button from '@mui/material/Button';

import './index.css';

const CardItem = ({ service, deleteService, editService }) => {
  const { title, description, id } = service;

  return (
    <>
      <Card className="card-container">
        <CardContent className="card-content">
          <Typography
            className={'MuiTypography--heading'}
            variant={'h6'}
            gutterBottom
          >
            {title}
          </Typography>
          <Typography
            className={'MuiTypography--subheading'}
            variant={'caption'}
          >
            {description}
          </Typography>
          <Divider className="card-divider" light />
          <Stack direction={{ xs: 'column', sm: 'row' }} spacing={1}>
            <Button
              variant="outlined"
              startIcon={<EditIcon />}
              onClick={() => editService(service)}
            >
              Editar
            </Button>
            <Button
              variant="outlined"
              startIcon={<DeleteIcon />}
              onClick={() => deleteService(id)}
            >
              Eliminar
            </Button>
          </Stack>
        </CardContent>
      </Card>
    </>
  );
};

export default CardItem;
