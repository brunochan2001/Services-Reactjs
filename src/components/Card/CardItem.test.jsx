import { render, screen } from '@testing-library/react';
import CardItem from './CardItem';

describe('Pruebas en el componente <CardItem/>', () => {
  const service = {
    title: 'Salud',
    description: 'Compra de medicamentos',
    id: 1
  };

  test('debe de hacer match con el snapshot', () => {
    const { container } = render(<CardItem service={service}></CardItem>);
    expect(container).toMatchSnapshot();
  });

  test('debe de mostrar el titulo y description indicado', () => {
    render(<CardItem service={service}></CardItem>);
    expect(screen.getByText(service.title)).toBeTruthy();
    expect(screen.getByText(service.description)).toBeTruthy();
  });
});
