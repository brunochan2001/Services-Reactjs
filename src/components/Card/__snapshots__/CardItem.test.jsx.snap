// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <CardItem/> debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiCard-root card-container MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiCardContent-root card-content"
    >
      <h6
        class="MuiTypography-root MuiTypography--heading MuiTypography-h6 MuiTypography-gutterBottom"
      >
        Salud
      </h6>
      <span
        class="MuiTypography-root MuiTypography--subheading MuiTypography-caption"
      >
        Compra de medicamentos
      </span>
      <hr
        class="MuiDivider-root card-divider MuiDivider-light"
      />
      <div
        class="css-o2cger-MuiStack-root"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="EditIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
              />
            </svg>
          </span>
          Editar
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="DeleteIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
          </span>
          Eliminar
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
